### PR TITLE
Calendar restructure

### PR DESCRIPTION
**Pull request by tsa87**

### CODE REVIEW

**Review by Vector-S — APPROVED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -11,230 +11,188 @@
...
 
-    return bestAvailabilityScore;
-  }
+      // Modify the the time ranges with penalties
+      if ((unavailableOptionalAttendeeCount > 0) || (unavailableMandatoryAttendeeCount > 0)) {
 
-  /* Return a list of TimeRange and associated Availability Score Pair from the previous TreeMap */
-  private ArrayList<Pair<TimeRange, Integer>> treeMapToTimeRangeScorePairList(TreeMap<Integer, Integer> timeCutoffs) {
+        Integer startIndex =
+          Collections.binarySearch(timeRangeAttendanceList, new TimeRangeAttendance(startTime, 0), c);
```
Comment (lipczak-g) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:109]: This code is somewhat hard to follow. I think it would be cleaner if you looked for the starting index and then craft the for loop to end when timeRange.startTime >= event.timeEnd.

Overall I think the complexity will be similar to the approach I proposed, although you look at it from the other side and use binary search instead of sort.
